ZuluContent/Skills/AnimalTaming.cs

```diff
@@ -1,338 +1,292 @@
+                    $"{from.Name} has angered the beast!");
+                creature.PlaySound(creature.GetAngerSound());
+                creature.Direction = creature.GetDirectionTo(from);
+                creature.BardEndTime = DateTime.Now;
+                creature.BardPacified = false;
+                creature.DoHarmful(from, true);
+                return true;
+            }
+
+            return false;
+        }
+
+        public static void CalmBeast(BaseCreature creature, Mobile from)
+        {
+            PacifyBeast(creature, from);
+            creature.PublicOverheadMessage(MessageType.Regular, 0x3B2, true,
+                $"{from.Name} has calmed the beast!");
+        }
+
+        public static void PacifyBeast(BaseCreature creature, Mobile from)
+        {
+            from.Combatant = null;
+            creature.Combatant = null;
+            creature.Warmode = false;
+            creature.Pacify(from, DateTime.Now + TimeSpan.FromSeconds(1.0));
+        }
+
+        private class InternalTarget : Target
+        {
+            public InternalTarget() : base(MaxDistance, false, TargetFlags.None)
+            {
+            }
+
+            protected override void OnTarget(Mobile from, object targeted)
+            {
+                from.RevealingAction();
+
+                if (!(targeted is BaseCreature))
+                {
+                    from.SendAsciiMessage(33, "You cannot tame this!");
+                    return;
+                }
+
+                BaseCreature creature = (BaseCreature) targeted;
+
+                if (!creature.Tamable)
+                {
+                    from.SendAsciiMessage(33, "You can't tame that!");
+                    return;
+                }
+
+                if (creature.Controlled)
+                {
+                    from.SendAsciiMessage(33, "That creature looks pretty tame already.");
+                    return;
+                }
+
+                var difficulty = creature.MinTameSkill;
+
+                if (from.Skills[SkillName.AnimalTaming].Value < difficulty)
+                {
+                    from.SendAsciiMessage(33, "You have no chance of taming this creature!");
+                    return;
+                }
+
+                if (m_BeingTamed.Contains(targeted))
+                {
+                    creature.PrivateOverheadMessage(MessageType.Regular, 0x3B2, 502802,
+                        from.NetState); // Someone else is already taming this.
+                    return;
+                }
+
+                difficulty += 10;
+
+                var timesPreviouslyTamed = creature.Owners.Count;
+
+                difficulty -= PrevTamedMinus * timesPreviouslyTamed;
+                if (difficulty < 1.0)
+                    difficulty = 1.0;
+
+                var calmingDifficulty = (int) difficulty + 10;
+
+                var isClassedRanger = (from as PlayerMobile)?.ZuluClass.Type == ZuluClassType.Ranger;
```

Comment:
Do you think you can move this to being an `IEnchantHook`?

It would be ideal if we could have all of the class bonus/mods inside of `ZuluClass`, e.g. you could create a

```csharp
        public void OnAnimalTamingAttempt(Mobile tamer, BaseCreature creature, ref double difficulty,
            ref double calmingDifficulty, ref double angerChance);
```

It would also allow us to create item enchantments later that target it too, e.g. "a shepard's crook of calming" => 25% less chance to anger during taming etc.

Check out the [ZuluClass hook for OnHeal](https://github.com/ZuluHotelAustralia/zuluhotel/blob/master/ZuluContent/Zulu/Engines/Classes/ZuluClass.cs#L346)

Reply:
Will check this out thx

Reply:
Made the change, thanks for pointing this out